source/gui/__init__.py

```diff
@@ -360,6 +356,20 @@ def onConfigProfilesCommand(self, evt):
 		self.postPopup()
 
 
+def safeAppExit():
+	"""
+	Ensures the app is exited by all the top windows being destroyed
+	"""

```

Comment:
Shouldn't this still try to detect if it is being called multiple times?

Reply:
I don't think that would be an issue anymore with the new behaviour. We can check for logging purposes but this should be able to be called multiple times safely as long as `window.Close` doesn't eventually call `safeAppExit`, which is the case for the mainFrame as it creates the `exitDialog`.